packages/contracts-bedrock/src/L2/OptimismSuperchainERC20.sol

```diff
@@ -141,4 +141,9 @@ contract OptimismSuperchainERC20 is SuperchainERC20, Initializable, ERC165 {
     function supportsInterface(bytes4 _interfaceId) public view virtual override returns (bool) {
         return _interfaceId == type(IOptimismSuperchainERC20).interfaceId || super.supportsInterface(_interfaceId);
     }
+
+    /// @notice Sets Permit2 contract's allowance at infinity.
+    function _givePermit2InfiniteAllowance() internal view virtual override returns (bool) {
+        return true;
```

Comment:
What is the rationale for this?

Reply:
This is due to the Solady update, to allow the Permit2 infinite allowance https://github.com/Vectorized/solady/pull/1093

Reply:
But why did we elect to enable it rather than keeping this `false`?

Reply:
This comes from the following issue: https://github.com/ethereum-optimism/optimism/issues/11899

And it is mostly from a UX perspective improvement: https://github.com/ethereum-optimism/optimism/issues/11880#issuecomment-2346807338

Reply:
OK, I'm good with this decision. I understand it as simply pre-approving the Permit2 contract for all users. Resolving